### PR TITLE
add MultiClusterIngress API for MCI feature

### PR DESCRIPTION
**Pull request by XiShanYongYe-Chang**
Signed-off-by: changzhen <changzhen5@huawei.com>

**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

As discussed in #1079, we plan to support Multi-Cluster Ingress. First of all, add the `MultiClusterIngress` API for the MCI feature.

**Which issue(s) this PR fixes**:
Part of #1079 

**Special notes for your reviewer**:

The API group of the MCI needs to be determined.

**Does this PR introduce a user-facing change?**:

```release-note
Add MultiClusterIngress API for MCI feature.
```

/cc @kevin-wangzefeng @RainbowMango @GitHubxsy

### CODE REVIEW

**Review by GitHubxsy — APPROVED** [review]
lgtm.
For the MultiClusterIngress API Group, I prefer net.karmada.io to work.karmada.io.

**Comment by XiShanYongYe-Chang** [comment]
> For the MultiClusterIngress API Group, I prefer net.karmada.io to work.karmada.io.

Done.

File: pkg/apis/networking/v1alpha1/ingress_types.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+import (
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:shortName=mcing
```
Comment (RainbowMango) [inline pkg/apis/networking/v1alpha1/ingress_types.go:11]: @GitHubxsy Are you ok with the short name `mcing`?
How do you think `mci`? 

By the way, the short name of Ingress in Kubernetes is `ing`.

Comment (GitHubxsy) [reply pkg/apis/networking/v1alpha1/ingress_types.go:11]: The `mci` would be more popular.

Comment (XiShanYongYe-Chang) [reply pkg/apis/networking/v1alpha1/ingress_types.go]: update it to `mci`.

**Comment by RainbowMango** [comment]
@GitHubxsy Should we wait for the proposal to move this PR forward?

**Comment by RainbowMango** [comment]
@XiShanYongYe-Chang Please update this PR as we just updated the k/k dependencies.

**Comment by XiShanYongYe-Chang** [comment]
CI looks not working!

**Comment by RainbowMango** [comment]
I can trigger failed workflows but can't for those not started.
Maybe you have to push again to trigger it.

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**